app/components/banner.tsx

```diff
@@ -1,37 +1,40 @@
 import { Trans } from "@lingui/react";
 import { Box } from "@mui/material";
+import { useMemo } from "react";
 
 import { HintYellow } from "@/components/hint";
 
 import { useChartState } from "../charts/shared/chart-state";
 
 /**
  * Banner to display errors or warnings for a chart. Each chart type can have specific warnings or errors depending on the data.
- *
- * @returns React.ReactNode | null
+ * Will only recalculate when the chart type changes.
  */
 export const Banner = () => {
   const state = useChartState();
 
-  switch (state.chartType) {
-    case "area":
-      const someValueIsNegative = state.allData.some(
-        (d) => +(state.getY(d) || 0) < 0
-      );
-      if (someValueIsNegative)
-        return (
-          <Box sx={{ mb: 4 }}>
-            <HintYellow iconName="datasetError" iconSize={64}>
-              <Trans id="dataset.error.negative-values">
-                Careful, this dataset contains negative values and might not
-                display correctly with this chart type.
-                <br />
-                <strong>Try using another chart type.</strong>
-              </Trans>
-            </HintYellow>
-          </Box>
+  const warning = useMemo(() => {
+    switch (state.chartType) {
+      case "area":
+        const someValueIsNegative = state.allData.some(
+          (d) => +(state.getY(d) || 0) < 0
         );
-    default:
-      return null;
-  }
+        if (someValueIsNegative) return <NegativeValueWarning />;
+      default:
+        return null;
+    }
+  }, [state.chartType]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Wondering why the eslint warning was removed ? I think it's fine to also pass state.allData and state.getY. You do not know whether getY would change the number of negative values for example.

Reply:
im just not sure what happens with the `useChartState()` that might cause it to be rebuilt.  I figure the only time we want the banner to rerender is when its first loaded when the chart type changes, but maybe I can try with exhaustive-deps